src/gpuarray_reduction.c

```diff
@@ -21,106 +21,613 @@
+                                           unsigned        reduxLen,
+                                           const unsigned* reduxList){
+	return GpuArray_reduction(GA_REDUCE_ANY,
+	                          dst,  NULL,   src, reduxLen, reduxList);
+}
+GPUARRAY_PUBLIC int  GpuArray_reduction   (ga_reduce_op    op,
+                                           GpuArray*       dst,
+                                           GpuArray*       dstArg,
+                                           const GpuArray* src,
+                                           unsigned        reduxLen,
+                                           const unsigned* reduxList){
+	redux_ctx  ctxSTACK = {op, dst, dstArg, src,
+	                       (int)reduxLen, (const int*)reduxList},
+	          *ctx      = &ctxSTACK;
+
+	return reduxCheckargs(ctx);
+}
+
+/**
+ * @brief Get an expression representing a suitable initialization value for
+ *        the given datatype and a sum-reduction operation.
+ *
+ * @param [in]  typecode  Typecode of the type whose initializer is to be
+ *                        requested.
+ * @param [out] property  A pointer to a string. On return it will be set to
+ *                        the initializer expression.
+ * @return Zero if successful; Non-zero if the datatype is not supported.
+ */
+
+static int   reduxGetSumInit               (int typecode, const char** property){
+	if(typecode == GA_POINTER ||
+	   typecode == GA_BUFFER){
+		return GA_UNSUPPORTED_ERROR;
+	}
+	*property = "0";
+	return GA_NO_ERROR;
+}
+
+/**
+ * @brief Get an expression representing a suitable initialization value for
+ *        the given datatype and a prod-reduction operation.
+ *
+ * @param [in]  typecode  Typecode of the type whose initializer is to be
+ *                        requested.
+ * @param [out] property  A pointer to a string. On return it will be set to
+ *                        the initializer expression.
+ * @return Zero if successful; Non-zero if the datatype is not supported.
+ */
+
+static int   reduxGetProdInit              (int typecode, const char** property){
+	if(typecode == GA_POINTER ||
+	   typecode == GA_BUFFER){
+		return GA_UNSUPPORTED_ERROR;
+	}
+	*property = "1";
+	return GA_NO_ERROR;
+}
+
+/**
+ * @brief Get an expression representing a suitable initialization value for
+ *        the given datatype and a max-reduction operation.
+ *
+ * @param [in]  typecode  Typecode of the type whose initializer is to be
+ *                        requested.
+ * @param [out] property  A pointer to a string. On return it will be set to
+ *                        the initializer expression.
+ * @return Zero if successful; Non-zero if the datatype is not supported.
+ */
+
+static int   reduxGetMinInit               (int typecode, const char** property){
+	switch(typecode){
+		case GA_BYTE2:
+		case GA_BYTE3:
+		case GA_BYTE4:
+		case GA_BYTE8:
+		case GA_BYTE16:
+		case GA_BYTE:               *property = "SCHAR_MIN"; break;
```

Comment:
This might be problematic for the densest `switch`es I have, since that will triple their size for no gain in readability. Are all these STYLE fixes you suggest something you feel increases readability, or are they simply the output of a soulless `lint`er?